src/System.Resources.Extensions/src/System/Resources/Extensions/PreserializedResourceWriter.cs

```diff
@@ -155,12 +155,14 @@ public void AddBinaryFormattedResource(string name, byte[] value, string? typeNa
                 // and reserializing when writing the resources.  We don't want to do that so instead
                 // we just omit the type.
                 typeName = UnknownObjectTypeName;
-
-                // ResourceReader will validate the type so we must use the new reader.
-                _requiresDeserializingResourceReader = true;
             }
 
             AddResourceData(name, typeName, new ResourceDataRecord(SerializationFormat.BinaryFormatter, value));
+
+            // Even though ResourceReader can handle BinaryFormatted resources, the resource may contain
+            // type names that were mangled by the ResXWriter's SerializationBinder, which we need to fix
+
+            _requiresDeserializingResourceReader = true;
```

Comment:
>_requiresDeserializingResourceReader [](start = 12, length = 36)

would be any perf concern with that?

Reply:
No: it’s a new dependency, but I believe it was being used anyway, since ResXDataNode wasn’t setting type in the resx for BinaryFormatted data https://github.com/dotnet/winforms/blob/d98947638f53765c79a2436cce2d9445e865a331/src/System.Windows.Forms/src/System/Resources/ResXDataNode.cs#L359

So MSBuild ended up always going down the path here that requires the new reader.